### PR TITLE
📖 Documentation for deploy-image plugin

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
Documentation for deploy-image plugin

### CODE REVIEW

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
@NikhilSharmaWe have the doc design for the plugin already.
What we will need is a documentation on how to use the plugin and etc in the website, see:
https://book.kubebuilder.io/plugins/plugins.html

Note that we have one page for each available plugin. So, we need will need a new doc very similar.

**Review by camilamacedo86 — COMMENTED** [review]
/hold 

Until we have the plugin merged but we can work alongside in the reviews 
Thank you for the PR.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:12]:
```suggestion
Se the project "project-v3-with-deploy-image" under the [testdata][testdata] directory of the Kubebuilder project to check an example of the scaffold produced with this plugin.
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:18]:
```suggestion
- If you are getting started with the [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/), Kubebuilder and [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) this plugin can help you
- If you are looking to Deploy and Manage an image (Operand) using  [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) and the tool the plugin will create an API/controller to be reconciled and achieve this goal
- If you are looking for examples and recommendations you might want to check out the scaffold done by this plugin 
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:24]:
```suggestion
kubebuilder create api --group example.com --version v1alpha1 --kind Memcached --image=memcached:1.6.15-alpine --image-container-command="memcached -m=64 modern -v" --image-container-port="11211" --plugins="deploy-image/v1-alpha"
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:32]:
```suggestion
```
Then, I think we can remove this one

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:45]:
```suggestion
```
Not yet. We did not implemented the tests yet

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:49]:
```suggestion
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:42]:
```suggestion
By calling create api with this plugin additionally to what is created and update by default we have the following customizations on top:
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:45]:
```suggestion
```
not yet implemented

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:44]:
```suggestion
- `controllers/*_controller.go` (the plugins scaffold the code implementation on the controller created)

```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:46]:
```suggestion
- `api/*_types.go` (scaffold the specs for the new api)

```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:47]:
```suggestion
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:48]:
```suggestion
- `config/samples/*_.yaml` (scaffold default values for its CR)

```

**Review by camilamacedo86 — COMMENTED** [review]
Great work @NikhilSharmaWe 🥇 
Following some small nits and suggestions.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (NikhilSharmaWe) [reply docs/book/src/plugins/deploy-image-plugin-v1-alpha.md]: Updated with suggested changes.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (FillZpp) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:40]: It will be `controllers/{group}/*_controller.go` when multigroup is enabled.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (FillZpp) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:41]: It will be `controllers/{group}/*_controller.go` when multigroup is enabled.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (FillZpp) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:42]: It should be `api/{version}/*_types.go`. And it will be `apis/{group}/{version}/*_types.go` when multigroup is enabled.

Comment (FillZpp) [reply docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:44]: Note that this is not fixed yet. It should be `api/<version>/*_types.go` instead of `api/*_types.go`  :)

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:27]: @NikhilSharmaWe we need to update here, see the command values ^

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [reply docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:42]: we can add this line as well ^ . good point

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:27]: @NikhilSharmaWe we pass the commands splitted by "," right?
Because we are splitting the string. We need to update this one here.

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:47]:
```suggestion
When multigroup is enabled
```suggestion
**When multigroup is enabled**

```
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (camilamacedo86) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:27]:
```suggestion
kubebuilder create api --group example.com --version v1alpha1 --kind Memcached --image=memcached:1.6.15-alpine --image-container-command="memcached,-m=64,modern,-v" --image-container-port="11211" --run-as-user="1001" --plugins="deploy-image/v1-alpha"

```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (varshaprasad96) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:24]:
```suggestion
Then, by using this plugin you can [create APIs][https://book.kubebuilder.io/cronjob-tutorial/gvks.html] informing the image (Operand) that you would like to deploy on the cluster. Note that you can optionally specify the command that could be used to initialize this container via the flag `--image-container-command` and the port  with `--image-container-port` flag. You can also specify `RunAsUser` value for the Security Context of the container via the flag `--run-as-user`., i.e:
```

File: docs/book/src/plugins/deploy-image-plugin-v1-alpha.md
Comment (varshaprasad96) [inline docs/book/src/plugins/deploy-image-plugin-v1-alpha.md:38]:
```suggestion
With the `create api` command of this plugin, in addition to the existing scaffolding, the following files are affected:
```

**Review by camilamacedo86 — APPROVED** [review]
/approved
/lgtm 

Great work 🥇

**Pull request merged**
